### ISSUE TITLE
Knowledge graph quality control: A survey

### ISSUE THREAD

**Issue by dreamingjudith** [dreamingjudith/paper-reading-list#13]
[Link](https://www.sciencedirect.com/science/article/pii/S2667325821001655)

# 3줄 요약

- Knowledge graph (이하 KG)의 품질이라는 게 뭔지 6가지 주요 Dimension으로 정의함
- KG를 구축하는 과정에서의 Quality control 소개
- 이미 구축된 KG의 품질을 향상하는 방법 제시

# 2. Evaluation dimensions of KG quality

## 2.1. Related framework and KG quality dimension analysis

- 이미 잘 알려진 프레임워크*들도 나름대로의 quality에 대한 dimension을 정의하고 있으나 겹치거나 비슷한 정의들이 많음
    - Zaveri et al.
    - Stvila et al.
- 적용 태스크의 요구사항이나 데이터 구조 등이 차이로 인해 불필요하고 중복되는 dimension도 아주 조금씩 존재

### 2.2.1. Accuracy

- 여러가지 정의
    - degree of correctness of facts in a KG
    - "the extent to which knowledge are correct, reliable, and certified free of error" - Wang et al.
- 평가하는 방법
    - gold standards and expert verification
    - functional dependency between the attributes of entities - Forber et al.
    - the ratios of inaccurate labels, annnotations, and classifications o entities - Lei et al.
- KG가 information retrieval이나 question answering 등에 사용되기 때문에 중요한 척도임

### 2.2.2. Consistency

- 정의: the degree to which the knowledge of a KG

![Untitled](https://user-images.githubusercontent.com/20284375/149776785-b3d63c74-9901-4dd6-80c9-dececfc109de.png)

- accuracy가 semantic correctness에 초점을 맞춘 데에 비해 consistency는 logical correctness 중심임.
    - 시간이 흘러버린 바람에 더이상 사실이 아닌 내용임에도 불구하고 logical correctness는 여전히 보장되기 때문에 consistency는 높은 경우
    - 그렇지만 high accuracy가 보장되는 fact는 self-consistent하기 때문에 high consistency가 보장됨
- 평가하는 방법
    - the proportion of entities that belong to disjoint classes as the assessment criteria - Hogan et al.
    - generating a set of manually-verified schema axioms - Zaveri et al.
    - consistency-based error detection이 KG accuracy도 상승시켰다는 경우도 있었음

### 2.2.3. Completeness

- 정의: the extent to which a KG covers the knowledge of interest
    - knowledge of interest라는 개념 자체가 상대적이기 때문에 본 개념도 상대적임
- 트리플렛을 이루는 entity attribute가 온전하면 이들을 explicitly 표현하는 관계도 온전한지를 보는 게 관건
- 평가 방법
    - the number of classes, entites, attributes, etc., for defining related indicators and scoring functions - Mendes et al.
    - nonparametric statistical techniques - Luggen et al.

### 2.2.4. Timeliness

- 몇 가지 정의
    - the degree to which knowledge is up-to-date
        - 예를 들어 매일 갱신되는 KG는 high timeliness를 가지고 있다
    - the time range since the last update         or the last verification of the statement - Naumann et al.
        - Frber et al.은 이 척도를 timeliness를 평가하는데 사용함
    - the ratio of time from the last update to the time from the first existence of knowledge
- 신기하게도 timeliness와 accuracy는 높은 관련성이 있다
    - 예를 들어 (Trump, IsPresidentOfthe, United Status) 라는 트리플렛이 생성 당시에는 최신이었을지 몰라도 지금은 accurate하지 않다는 것

### 2.2.5. Trustworthiness

- 정의: the degree of objectivity, authority, and verifiability of a KG
- 다소 주관적일 수 있는 개념이기 때문에 다른 척도들에 많은 영향을 받는 편
    - accuracy, timeliness나 consistency를 높이는 것만으로도 KG의 공신력을 올리는 거기 때문에 trustworthiness를 올려주는 것과 마찬가지
- 소스 선택이나 구축 방법에 영향을 많이 받음
- 평가 방법
    - decision network of knowledge source graphs for trustworthiness evaluation - Gamble
    - assessing KG accuracy

### 2.2.6. Availability

- 정의: the extent to which knowledge is convenient to use
- 정량적으로 평가하기 어렵기 때문에 세 부분으로 나눌 수 있음
    - whether a KG is in a machine-readable form or can be downloaded // for machines
    - the ability to support different languages, usage of technical queries // for users
    - the speed of data response, degree of interconnection between entities with similar properties, and form of knowledge representation // for fast retrieval of knowledge
- availability는 다른 evaluation dimensions들과는 달리 평가나 향상을 위해 KG의 디자인, 아키텍처 등등을 살펴봐야 함

![Untitled 1](https://user-images.githubusercontent.com/20284375/149776837-07df497a-55e3-4822-bd62-2ac511f5b0be.png)

# 3. Quality control during KG construction

## 3.1. Knowledge source selection

- 주로 credibility, relevance에 영향을 미침
    - crediblitiy: the degree to which knowledge sources are convincing
    - relevance: the degree to which knowledge sources are related to the knowledge of interest
- 따라 평가 시에도 두 가지의 evaluation dimension의 관점에 따라 나뉘어 평가하게 됨
    - designing credibility evaluation methods of knowledge sources to select high-credibility knowledge source
    - improving the relevance of knowledge sources through noise filtering

### 3.1.1. Online knowledge source selection

- Credibility selection: generally evaluated from the perspectives of importance and content accuracy
    - Web page importance
        - link-based information in the web structurue로 주로 평가되는데 이를 위해 사용하는 게 propagation mechanism
        - PageRank: using **PR value**
        - improved propagation-based algorithms의 경우 construct a network structure with links betwen different web pages or data and evaluate the credibility of target knowledge sources through propagation of the designed metrics
    - Accuracy of web page content
        - distant supervision 기법이 주로 이용되며
        - external knowledge base와 비교해서 웹 페이지의 콘텐츠를 평가하는 방법이 주로 사용됨
        - textual content, link structure, web page design 등 웹페이지를 이루는 다양한 요소들을 통해 신뢰도를 측정하는 방법이 제안되기도 함 - Olteanu et al.
- Relevance enhancing (or noise filtering)
    - HTML tag-based
    - pure text-targeted
        - word frequency를 기반으로 함
    - feature design-based

### 3.1.2. Crowdsourcing knowledge source selection

- “spammer”로 인해 주로 발생하는 false labels을 줄이는 것이 필요
- false labels을 줄이는 방법
    - majority voting
        - 정상적인 사람들 사이에 소수의 “스패머”가 섞여있을 때 효과
        - “스패머”를 거를 수는 없기 때문에 1차적으로 머신러닝으로 스패머를 거른 다음 나머지에 대해서 majority voting을 적용하는 경우도 있음
        - 어노테이터 간의 레이블 편차가 심할 경우 문제가 생김
    - the expectation-maximization(EM) algorithm
    - EM algorithm과 같이 쓰는 통계 모델 e.g. empirical Bayesian algorithm
    - Sequence tagging에 사용되는 알고리즘
        - Hidden Markov model
        - recurrent neural network
        - long short-term memory
    - Bi-LSTM과 같은 신경망 모델

### 3.1.3. Offline knowledge source selection

- web page text를 대상으로 하는 relevance control과 유사

### 3.1.4. Database knowledge source selection

- distant supervision

### 3.1.5. Experts as a knowledge source

- 직접적으로 평가할 수 없으나 간혹 distant supervision을 적용하기도 함
- manual inspection

## 3.2. Knowledge extraction

- Named entity recognition (NER) —> Entity relation extraction (ERE)

### 3.2.1 Quality control of NER

- 영향을 미치는 evaluation dimensions
    - Error-recognized entities는 accuracy, trustworthiness를 악화시킴
    - Unrecognized entities는 completeness에 영향을 미침
- Quality control
    - of pattern-based NER methods
        - type checking method
        - instance confidence assessment method
        - graph-based algorithm
        - removing “drift points”
            - 이 경우 여전히 multi-concept instances는 걸러내지 못 한다
        - coupled pattern learner (CPL) by asking people to take part in the knowledge acquisition process
    - of learning-based NER methods
        - feature and model design
        : makes predictions on entity boundaries by learning features
            - traditional ML system —> feature design
                - feature space: 11 orthographic features, entity trigger words, gazetteer words // HMM
                - + capitalization, first occurrence flag of words, occurrence flag with a known last name and first names from dictionary // Maximum entropy
                - neighborhood words
                - suffix features
                - word embedding features
            - NN learning system —> both feature and model design
                - LSTM
                - Bi-LSTM + CRF
            - Cross-domain NER: NN learning system + transfer learning
                - multi-task learning
                - fine-tuning
                - share-private
                - knowledge distilling
        - quality of the tagged corpus
            - distant supervision methods
                - attention mechanism
                - partially annotated CRFs
                - heuristic rules
            - weakly supervised methods (occurrence popularity-based methods)

### 3.2.2 Quality control of ERE

- Pattern-based ERE methods
    - using expert rules requiring significant manual work for rule design
    - Semi-supervised ERE method e.g. bootstrapping
- Learning-based ERE methods
    - Traditional ML systems
        - More refined design of the feature space
        - Kernel methods e.g. syntactic tree kernel method
    - NN systems
        - CNN
        - Attention mechanism
        - attention + distant supervision
        - sequential and tree-structured LSTM + attention based on a dependency tree
        - a novel selective gate framework (self-attention + pooling-equipped gate)
    - Unsupervised ERE methods (based on clustering, for low-quality labeled data)
- Joint extraction of relations and entities
    - sharing the training parameters of the NNs of NER and ERE
    - extending the information of word tags: joint extraction task —> sequence tag prediction task
    - Dissolving relationship overlap using multi-head selection task (w/ Bi-LSTM-CRF)
    - Seq2Seq models (CopyRE, CopyMTL)

### 3.3.1. Entity linking

- Entity disambiguation —> similarity analysis
    - bag-of-words
    - semantic models
    - soclai network models
    - encyclopedia knowledge models
- Co-referential resolution
    - syntactic analysis
    - traditional ML methods e.g. decision tree, Dempster-Shafer model
    - clustering-based methods
    - neural network
        - end-to-end models

### 3.3.2. Knowledge base merging

- entity linking과는 달리 entity와 relation 모두 연결하는 작업
- completeness, accuracy, trustworthiness
- general goal: to calculate the similarity between various entities (relations) and choose the most similar one for linkage
    - entity, entity attribute, and relation embedding using deep NN-based encoders
    - similarity calculation method based on semantic labels
- Relation database와 RDF format의 상호 변환을 위해 두 가지 표준 제안 by W3C
    - direct mapping
    - R2RML

# 4. Enhancement of KG quality

### 4.1.1. Error detection of relations

- Internal knowledge-based detection methods
    - Path ranking algorithm
    - Using entity type feature
    - Statistical distribution-based methods
- Distant supervision for external data

### 4.1.2. Error detection of entity type

- Relations have more tendency to be wrong in KGs compared to entity type (by Paulheim and Bizer)
- associated features, statistical distribution

### 4.1.3. Error detection of attributes

- Outlier detection
    - 이 때 natural outlier는 보존해야 함 왜냐면 특정 국가의 인구 수나 어떤 산의 높이 등 자연적 아웃라이어가 많기 때문
    - unsupervised numerical outlier detection using interquartile range and semantic grouping by Wienand and Paulheim
    - + combining the outcomes of multiple independent outlier detections by Fleischhacker et al.
- Multi-attribute consistency
    - combining the information of entities and discovering inconsistencies
    - Sequential dependencies to express relationships between ordered attributes and find the wrong attributes by Golab et al.
    - metric functional dependencies
    - arithmetic operations between numerical attributes
        - numerical function dependency relationship for detecting wrong numerical attributes by Fan et al.
        - probabilistic framework for detecting errors in the numerical attributes of an entity by Li et al.

### 4.2.1 Global update

- time span을 정해놓고 그걸 넘어가는 항목들에 대해서는 knowledge extraction, knowledge fusion과 같은 acquisition 수행
- KG를 재구축하는 것과 비슷한 리소스가 사용되기 때문에 작은 규모나 업데이트가 적은 KG에 적합

### 4.2.2. Local update

- communication between the knowledge source and KG
- to predict the knowledge updates
    - Statistical methods: Bayesian inference, stochastic modeling
    - Estimation of the update frequency of entities in the knowledge source based on the incomplete update history b Cho et al.
- Deep learning methods
    - predictive model that uses historical update information and semantic information of an entity as features by Liang et al.
    - time-ware knowledge update prediciton models by Jiang et al.
        - temporal order information and imposes time sequence constraints on the geometric structure of the embedded space
        - based on integer linear programming and considers global time constraints in reasoning

### 4.3.1. Completion of relations

- probabilistic graphical models
- deep learning methods
    - learn the latent factor representation of entities and relations for relation prediction
    - TransE, TransH, TransR, TransD, TransSparse, ...
- tensor factorization
    - to represent a KG as a 3D tensor completion problem
    - RESCAL, TuckER, ...
- 그러나 사실 위에서 소개된 모델들 모두 한계점을 가지고 있음
    - general dataset에서만 실험
    - KG 안에 있지 않은 entity나 relation을 예측하는데는 실패
    - Open-workd KGC task라고 빠르게 변화가 필요한 KG에 적용하기 위해 새로 제안된 태스크가 있으나 아직 연구가 필요한 상황

### 4.3.2. Completion of entity types

- heuristic probability model (e.g. SDtype: type inference mechanism - 걍 주변에 연결된 엔티티들의 타입 개수를 센 다음에 frequency가 많은 쪽으로 가는 것)
- learning-based methods

### 4.3.3. Completion of attributes

- 다른 태스크들에 비해서 어려운 이유
    - lack of attributes of entities가 implicit하기 때문
    - the classification method may be difficult to apply widely due to the different hierarchical structures of different entities
    - numerical attributes의 경우 accuracy requirements가 요구되는 regression methods가 필요
- lack of attributes를 판단하는 방법 by Razniewski et al.
    - referring to similar entities
    - according to the amount of access of the entity
    - according to the growth pattern of an entity’s attribute
    - through the pattern matching methods
- Probase+ by Liang et al.
    - Collaborate filtering 사용
    - 비슷한 엔티티들끼리는 비슷한 upper and lower level이 있다고 가정
    - hypernym, hyponym이 이러한 방법을 통해 missing attributes 잘 찾을 수 있음
- 일단 missing attributes를 찾고 나면 crowdsourcing이나 distant supervision과 같은 external completion approach 적용 가능

## 4.4. Consistency enhancement

- disjointness axioms을 통해 inconsistent knowledge를 찾아내는 방법
- top-level ontologies e.g. *DOLCE*
- knowledge reasoning
    - 그러나 사실 reasoning을 할 수 있을 정도로 ontology가 충분히 rich하지 않은 경우가 대부분이기 때문에 enriching ontologies와 함께 쓰이는 경우가 대부분

## 4.5. Availability enhancement

- 요지는 large-scale KG를 전체 다 다운받지 않더라도 필요한 subgraph만 로딩할 수 있게 하는 것
- query load를 분산시키는 방법
- query result를 최적화하는 방법
- to assist users by expanding their query with intuitive suggestions
- SAKQ

## 4.6. Trustworthiness enhancement

- 주로 온라인 정보를 통해 구축되는 KG의 특성상 세 가지 측면에서 분석 가능
    - knowledge source
        - knowledge fusion methods에서 소개된 방법들을 통해 퀄리티 확보
    - information contained in a KG
        - internal enhancement methods, knowledge fusion methods
    - KG spreading routes
        - analysis of social network of users for external information 정도 적용해볼 수 있을 거 같다?

# 5. KG support for AI methods

- Question answering의 사례
    - question과 candidate answers를 distributed representation으로 encode하고 different similarity calculation을 통해 가장 적절한 정답을 선택하는 방식
    - 이 때 필요한 KG의 속성
        - good locgical consistency
        - good semantic accuracy of relations
        - good structural accuracy of KG
    - KG가 제공할 수 있는 것: semantic structure of knowledge through the relation connections between entities, which can be used to infer indirect facts
